indexer/src/processing.rs

```diff
@@ -13,63 +14,31 @@ use apibara_core::{
 };
 use apibara_sdk::{DataMessage, DataStream};
 use chrono::{DateTime, Utc};
-use tokio::time::{sleep, Duration};
 use tokio_stream::StreamExt;
 
-// pub async fn run_processing_loop(
-//     data_stream: &mut DataStream<Filter, Block>,
-//     conf: &config::Config,
-//     state: &Arc<AppState>,
-// ) {
-//     loop {
-//         let result = process_data_stream(data_stream, conf, state).await;
-
-//         if let Err(e) = result {
-//             println!("Disconnected: {:?}", e);
-//             sleep(Duration::from_secs(5)).await;
-//             continue;
-//         }
-
-//         // If processing completed successfully, break out of the loop
-//         // break;
-//     }
-// }
-
 pub async fn process_data_stream(
     data_stream: &mut DataStream<Filter, Block>,
     conf: &config::Config,
     state: &Arc<AppState>,
 ) -> Result<()> {
-    // while let Some(message) = data_stream.try_next().await.unwrap() {
-    loop {
-        match data_stream.try_next().await {
-            Ok(Some(message)) => match message {
-                DataMessage::Data {
-                    cursor: _,
-                    end_cursor: _,
-                    finality,
-                    batch,
-                } => {
-                    if finality != DataFinality::DataStatusFinalized {
-                        println!("shutting down");
-                        break;
-                    }
-
-                    for block in batch {
-                        process_block(&conf, &state, block).await?;
-                    }
+    while let Some(message) = data_stream.try_next().await.unwrap() {
```

Comment:
please try to remove this unwrap, the connection will probably get closed at some point, you can check https://github.com/starknet-id/analysor